app/models/user.rb

```diff
@@ -1,8 +1,18 @@
 class User < ApplicationRecord
   has_secure_password
 
+  has_many :party_users, dependent: :destroy
+  has_many :parties, through: :party_users
   has_many :friendships, dependent: :destroy
   has_many :friends, through: :friendships
 
   validates :email, uniqueness: true, presence: true
+
+  def any_parties?
+    !party_users.empty?
+  end
+
+  def party_status(party_id)
+    party_users.where(party_id: party_id).pluck(:attendee_type).first
```

Comment:
returns 'host' or 'guest' for a user for a specific party (used in the dashboard view)

Reply:
I would like to look at this together to ensure it is optimized. This came up a bit during the M2 final project, and using AR.